### PR TITLE
Use Kokkos::bit_cast if available

### PR DESCRIPTION
**Pull request by aprokop**

### CODE REVIEW

File: src/details/ArborX_MinimumSpanningTree.hpp [hunk 1]
```diff
@@ -624,7 +625,12 @@ void computeParents(ExecutionSpace const &space, Edges const &edges,
         // Comparison of weights as ints is the same as their comparison as
         // floats as long as they are positive and are not NaNs or inf
         static_assert(sizeof(int) == sizeof(float));
-        keys(e) = (key << shift) + reinterpret_cast<int const &>(edge.weight);
+        keys(e) = (key << shift) +
+#if KOKKOS_VERSION >= 40099
```
Comment (dalg24) [inline src/details/ArborX_MinimumSpanningTree.hpp:629]: No that's too brittle that would break anyone on develop who didn't pull this morning.

**Review by dalg24 — CHANGES_REQUESTED** [review]

File: src/details/ArborX_MinimumSpanningTree.hpp (hunk 1 above)
Comment (dalg24) [reply src/details/ArborX_MinimumSpanningTree.hpp:629]: OK if you use 4.1

**Review by masterleinad — COMMENTED** [review]
I am not a big fan of testing for a version still in development. I don't see a reason why we can't at least wait for the release.

**Comment by aprokop** [comment]
> I am not a big fan of testing for a version still in development. I don't see a reason why we can't at least wait for the release.

We can. I wanted to add essentially a reminder so that we don't forget, but may have been too aggressive. I'm fine with either protecting by 4.1 or not doing it.

**Review by dalg24 — APPROVED** [review]

**Review by masterleinad — APPROVED** [review]

**Pull request merged**